sklearn/linear_model/cd_fast.pyx

```diff
@@ -510,7 +510,7 @@ def enet_coordinate_descent_gram(floating[::1] w,
     cdef floating d_w_tol = tol
     cdef floating dual_norm_XtA
     cdef unsigned int ii
-    cdef unsigned int n_iter = 0
+    cdef int n_iter = 0
```

Comment:
Do you remember what was the warning? uint seems like the right type for `n_iter`..

Reply:
`n_iter` is used to iterate later:

```py
for n_iter in range(max_iter):
    ...
```

where `max_iter` is an int. When `n_iter` is a unsigned int we get warning similar to this:

```
warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
         for (__pyx_t_10 = 0; __pyx_t_10 < __pyx_t_29; __pyx_t_10+=1) {

```

Reply:
Alternatively we can cast `max_iter` to an unsigned int.

Reply:
Casting seems like a bad idea, there isn't any checking of `max_iter` when `enet_path` calls `enet_coordinate_descent`.